optuna/storages/_cached_storage.py

```diff
@@ -180,6 +181,11 @@ def create_new_trial(self, study_id: int, template_trial: Optional[FrozenTrial]
                 study.owned_or_finished_trial_ids.add(frozen_trial._trial_id)
         return trial_id
 
+    @deprecated(
+        "3.0.0",
+        "4.0.0",
+        text="Use :func:`~optuna.storages._CachedStorage.set_trial_state_values` instead.",
+    )
```

Comment:
Since we have not make `CachedStorage` and `InMemoryStorage` public, we can remove unnecessary methods. So, could you remove `set_trial_state()` and `set_trial_values()` from `CachedStorage` and `InMemoryStorage`?

Reply:
Ok, fixed. I also remove remove `abstract` from `set_trial_state()` and `set_trial_values()` of `BaseStorage` class.